baybe/utils/sampling_algorithms.py

```
@@ -13,6 +13,7 @@ def farthest_point_sampling(
     n_samples: int = 1,
     initialization: Literal["farthest", "random"] = "farthest",
     random_tie_break: bool = True,
+    start_idx: int | None = None,
```

This breaks the existing interface! For example, if someone calls 
```python
farthest_point_sampling(points, initialization="random", start_idx=0)
```
then their `initialization` argument would be silently dropped

---

Also, what's the motivation for the new argument in the first place? Is it related to fpsample?

---

Just as FYI: We discussed this with Martin here: https://github.com/emdgroup/baybe/pull/547#discussion_r2132162255

---

I think your answer belongs to a different thread, could that be?

---

Ah I see - thanks for flagging that. The motivation behind the new start_idx argument is to control the starting index for both the original internal FPS implementation and for fpsample. I added this for cross-checking behavior in one of the tests, this way it ensures consistent starting points across implementations.

That said, it shouldn’t override existing initialization logic or silently drop user-specified arguments. Thanks for the catch! I’ll update the implementation and please et me know if you have any suggestions on how you’d prefer it to be structured!

---

reopening this for adrian to resolve
fyi there is now a raise in case the start index is specified but the initalization method is not compatible with that

---

ok I can see that

how would the collection of ints be used in the case you mentioned? There we need exclusion of points (that were already selected in previous iterations) not selecting them as starting indices (again and again in each iteration)

---

Not sure what you mean, but perhaps let me share my view:
1. First, we need to decide on what the return of the fps utility should look like: should the initialization points be included in the dataframe or not? since the `literal` modes include their first points, I think they should be part for consistency
1. Then, the recommender calling the method would simply pass the searchspace + the already selected points from the measurements dataset, request M+N points (where M is the size of the initial data set) and then drop the first M values from the return.

--> doesn't that account for everything?

---

it makes sense to me, one complexity that that simple workflow doesnt account for is that ingested datapoints might not correspond exactly to recommended ones

so this is clearly something that needs to be addressed outside of this PR, for here simply allowing the collection should be sufficient and I'll reply once thats added